app/jobs/set_catkeys_and_barcodes_job.rb

```diff
@@ -23,7 +23,7 @@ def perform(bulk_action_id, params)
       update_druids.each_with_index do |current_druid, i|
         cocina_object = Repository.find(current_druid)
         args = {}
-        args[:catkeys] = catkeys[i] if catkeys
+        args[:catkeys] = (catkeys[i] || []) if catkeys
         args[:barcode] = barcodes[i] if barcodes
```

Comment:
this comes in as nil if now value is set, we want it to be an empty array in that case